### PR TITLE
Save to Favorites

### PR DESCRIPTION
**Pull request by nhuy703**
Added save to favorite function
Fixed sign in & sign up UI

### CODE REVIEW

**Review by hthm00 — CHANGES_REQUESTED** [review]
- Save designs always show 2 items
Screenshot 2024-03-16 at 6 05 15 PM

- App crashes when trying to favorite an item without login
Screenshot 2024-03-16 at 6 07 17 PM

File: DesignScape/DesignScape/View/Product/FavoritesView.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    @StateObject var productViewModel = ProductViewModel()
+    @StateObject var viewModel = UserManager()
+    @State private var productUids: [String] = []
+    @State private var products: [Product] = []
+    var body: some View {
+        VStack(alignment: .leading) {
+            LazyVGrid(columns: [GridItem(.flexible(), spacing: 16),
+                                GridItem(.flexible(), spacing: 16)], spacing: 16) {
+                ForEach(products, id: \.self) { product in
+                    if let product = productViewModel.product {
```
Comment (hthm00) [inline DesignScape/DesignScape/View/Product/FavoritesView.swift:21]: this line cause repeated favorite items bug. see below
Screenshot 2024-04-02 at 1 33 20 PM

**Review by hthm00 — CHANGES_REQUESTED** [review]

File: DesignScape/DesignScape/View/Product/FavoritesView.swift (hunk 1 above)
Comment (nhuy703) [reply DesignScape/DesignScape/View/Product/FavoritesView.swift:21]: I fixed it hehe

**Review by hthm00 — APPROVED** [review]
Excellent work. I am so proud of you!

**Pull request merged**